### PR TITLE
Fix Site name & form factor in timeline

### PR DESCRIPTION
**Pull request by PoulpiFr**

### CODE REVIEW

File: app/angular/src/core/datamart/users/ViewOneController.js [hunk 1]
```diff
@@ -50,6 +50,8 @@ define(['./module', 'moment-duration-format'], function (module) {
 
       function retrieveSiteIdAndAppIdFromTimelines(activities) {
 
+        var activities = $scope.timelines;
```
Comment (dduponchel) [inline app/angular/src/core/datamart/users/ViewOneController.js]: `activities` is already computed and (almost) chained to this function. You should fix `waitForDevices` instead.

File: app/angular/src/core/datamart/timeline/timeline.js [hunk 2]
```diff
@@ -16,7 +16,13 @@ define(['./module'], function (module) {
     return decorateFilter;
   }]);
 
-  module.directive("mcsTimelineEventKey", [function () {
+  module.filter('isNotEmpty', [function() {
```
Comment (dduponchel) [inline app/angular/src/core/datamart/timeline/timeline.js]: An empty `timelineActivity.$topics` isn't always `{}`: the first timeline I checked returned `"$topics": null` and the page displayed an empty "Topics:" section.

**Comment by PoulpiFr** [comment]
@dduponchel > Fixed!

**Comment by PoulpiFr** [comment]
This PR could really make my life more beautiful =)

Having to look in the API to check on which site a User has gone on the timeline is a bit painful.

**Pull request merged**